pyrogram/client/client.py

```diff
@@ -183,16 +186,18 @@ def start(self):
             :class:`pyrogram.Error`
         """
         self.load_config()
-        self.load_session(self.session_name)
-
-        self.session = Session(
-            self.dc_id,
-            self.test_mode,
-            self.proxy,
-            self.auth_key,
-            self.api_key.api_id,
-            client=self
-        )
+        if callable(self.load_session_hook):
+            self.load_session_hook(self)
+        else:
+            self.load_session(self.session_name)
```

Comment:
Not sure if you indented it wrongly, but this piece of code is always needed, here is where Pyrogram creates a new connection to the server.